src/main/kotlin/com/openlattice/shuttle/controllers/ShuttleController.kt

```diff
@@ -0,0 +1,108 @@
+package com.openlattice.shuttle.controllers
+
+import com.codahale.metrics.annotation.Timed
+import com.openlattice.authorization.AuthorizationManager
+import com.openlattice.authorization.AuthorizingComponent
+import com.openlattice.shuttle.*
+import com.openlattice.shuttle.api.*
+import com.openlattice.shuttle.Integration
+import com.openlattice.shuttle.IntegrationJob
+import com.openlattice.shuttle.IntegrationStatus
+import com.openlattice.shuttle.IntegrationUpdate
+import org.springframework.web.bind.annotation.*
+import java.util.*
+import javax.inject.Inject
+
+@RestController
+@RequestMapping(CONTROLLER)
+class ShuttleController : ShuttleApi, AuthorizingComponent {
+
+    @Inject
+    private lateinit var integrationService: IntegrationService
+
+    @Inject
+    private lateinit var authorizationManager: AuthorizationManager
+
+    @Timed
+    @GetMapping(path = [INTEGRATION_NAME_PATH + INTEGRATION_KEY_PATH])
+    override fun enqueueIntegration(
+            @PathVariable(INTEGRATION_NAME) integrationName: String,
+            @PathVariable(INTEGRATION_KEY) integrationKey: UUID
+    ): UUID {
+        val normalizedName = normalizeIntegrationName(integrationName)
```

Comment:
does this not need admin access?

Reply:
So I believe the idea is that the context from which the api is called is not always going to have ol auths associated with it, and the integrationKey (which is created when the integration object is created, which does require admin access) is the authorization for running the integration.

I may not have explained this perfectly.